tests/algorithms/test_algorithm_resumption.py

```diff
@@ -0,0 +1,274 @@
+        'channels_last': 'Only for CNNs',
+        'colout': 'Only for images',
+        'cutmix': 'Only for images',
+        'cutout': 'Only for images',
+        'squeeze_excite': 'Not defined for language model transformers',
+        'mixup': 'Only defined for images',
+        'randaugment': 'Requires PIL dataset to test.',
+        'augmix': 'Required PIL dataset to test.',
+        'progressive_resizing': 'Only defined for images',
+        'stochastic_depth': 'Only applies to ResNets.',
+        'no_op_model': 'Not compatible with this model.'
+    }
+}
+
+
+@pytest.mark.timeout(180)
+@pytest.mark.parametrize("world_size", [
+    pytest.param(1),
+    pytest.param(2, marks=pytest.mark.world_size(2)),
+])
+@pytest.mark.parametrize("device_hparams,deepspeed_enabled,zero_stage", [
+    pytest.param(CPUDeviceHparams(), False, None, id="cpu-ddp"),
+    pytest.param(GPUDeviceHparams(), False, None, id="gpu-ddp", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 0, id="deepspeed-zero0", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 1, id="deepspeed-zero1", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 2, id="deepspeed-zero2", marks=pytest.mark.gpu),
+])
+@pytest.mark.parametrize(
+    "seed,save_interval,save_filename,resume_file,final_checkpoint",
+    [
+        [None, "1ep", "ep{epoch}-rank{rank}", "ep2-rank{rank}", "latest-rank{rank}"
+        ],  # test randomized seed saving and symlinking
+        [42, "1ep", "ep{epoch}-rank{rank}", "ep2-rank{rank}", "ep4-rank{rank}"],  # test save at epoch end
+        # [42, "1ep", "ep{epoch}-rank{rank}.tgz", "ep2-rank{rank}.tgz", "ep4-rank{rank}.tgz"
+        # ],  # test tarball with compression
+        # [42, "2ba", "ba{batch}-rank{rank}", "ba10-rank{rank}", "ba24-rank{rank}"],  # test save batch in partial epoch
+        # [42, "1ba", "ba{batch}-rank{rank}", "ba17-rank{rank}", "ba24-rank{rank}"],  # test save final batch in epoch
+        # [42, "1ba", "ba{batch}-rank{rank}", "ba19-rank{rank}", "ba24-rank{rank}"
+        # ],  # test save batch after complete epoch
+    ],
+)
+# @pytest.mark.parametrize("model_name", ["default", "resnet50_synthetic", "gpt2_52m"])
+@pytest.mark.parametrize("model_name", ["resnet50_synthetic"])
+@pytest.mark.parametrize("algorithm", algorithms_registry.items(), ids=algorithms_registry.keys())
+def test_algorithm_resumption(device_hparams: DeviceHparams, world_size: int, deepspeed_enabled: bool,
+                              zero_stage: Optional[int], save_interval: str, save_filename: str, resume_file: str,
+                              final_checkpoint: str, seed: Optional[int], model_name: str,
+                              algorithm: Tuple[str, Type[AlgorithmHparams]], dummy_model_hparams: ModelHparams,
+                              dummy_train_dataset_hparams: DatasetHparams, dummy_val_dataset_hparams: DatasetHparams,
+                              tmpdir: py.path.local):
```

Comment:
Use the new fixture :)

```suggestion
                              self_destructing_tmp: py.path.local):
```

Can you also update `test_checkpoint` to use this fixture?